bin/release.sh

```diff
@@ -79,9 +91,49 @@ update_version() {
         fi
     else
         status "Updating" "$crate from $curr_version to $version"
-        sed -i \
-            "/\[package\]/,/\[.*dependencies\]/{s/^version = \"$curr_version\"/version = \"$version\"/}" \
-            "$cargo_toml"
+        cargo set-version -p $crate $version
+    fi
+
+    # If we're releasing console-api, we need to update its version in
+    # the other crates in the workspace too.
+    if [[ "$crate" == "console-api" ]]; then
+        local cargo_upgrade=(cargo upgrade --offline -p console-api@$version)
+        if [[ "$verbose" ]]; then
+            cargo_upgrade+=("$verbose")
+        fi
+        
+        "${cargo_upgrade[@]}"
+        files+=($(ls Cargo.lock */Cargo.toml))
+    fi
+}
+
+commit() {
+    if ! [[ -z "${1+1}" ]] && [[ "$1" == "--amend" ]]; then
+        status "Amending" "release commit"
+        amend="$1"
+    else
+        status "Creating" "release commit"
+        amend=""
+    fi
+
+    # Prepare a commit message including the changelog from just this release.
+    tmp_dir=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')
```

Comment:
let's use a name for the tempdir that actually describes what it's for?

```suggestion
    tmp_dir=$(mktemp -d 2>/dev/null || mktemp -d -t 'tokio-console-release')
```

Reply:
Oops. I might have copied this straight from SO. 😅